### PR TITLE
[ci] Add path filter for CodeQL

### PR DESCRIPTION
**Pull request by zhongjiajie**
Add new rule for CodeQL, All `src` directory
except `pydolphinscheduler/src`

### CODE REVIEW

File: .github/workflows/codeql.yaml [hunk 1]
```diff
@@ -29,8 +29,24 @@ concurrency:
...
+    steps:
+      - uses: actions/checkout@v2
+      - uses: dorny/paths-filter@b2feaf19c27470162a626bd6fa8438ae5b263721
+        id: filter
+        with:
+          # All `src` directory except `pydolphinscheduler/src`
+          filters: |
+            not-ignore:
+              - 'src/**'
+              - '!(pydolphinscheduler/src/**)'
```
Comment (zhongjiajie) [inline .github/workflows/codeql.yaml:46]: I am not sure whether it work or not

File: .github/workflows/codeql.yaml [hunk 2]
```diff
@@ -29,8 +29,24 @@ concurrency:
   cancel-in-progress: true
 
 jobs:
+  paths-filter:
```
Comment (kezhenxu94) [inline .github/workflows/codeql.yaml:32]: This doesn't work in push event, when pushing to dev branch, CodeQL must be run and it acts as a basis for those runs in pull requests

Comment (zhongjiajie) [reply .github/workflows/codeql.yaml:32]: you mean we should also run CodeQL although we also have docs change in our PR?

Comment (kezhenxu94) [reply .github/workflows/codeql.yaml:32]: > you mean we should also run CodeQL although we also have docs change in our PR?

I mean we need to always run CodeQL in dev branch.

**Pull request closed without merging**